mne/decoding/time_gen.py

```diff
@@ -702,7 +765,12 @@ def _predict(X, estimators, is_single_time_sample):
     # Compute prediction for each sub-estimator (i.e. per fold)
     # if independent, estimators = all folds
     for fold, clf in enumerate(estimators):
-        _y_pred = clf.predict(X)
+        if predict_function == 'predict':
+            _y_pred = clf.predict(X)
+        elif predict_function == 'predict_proba':
+            _y_pred = clf.predict_proba(X)
+        elif predict_function == 'decision_function':
+            _y_pred = clf.decision_function(X)
```

Comment:
absolutely.

Is `predict_method` is sufficiently explicit? We already have
`predict_mode`.

We'll still need a test at the beginning of `gat.predict()` so that if the
user passes a function that the estimator doesn't have, it gets an error
before it runs in parallel.

On 16 January 2016 at 09:24, jona-sassenhagen <notifications@github.com>
wrote:

> In mne/decoding/time_gen.py
> <https://github.com/mne-tools/mne-python/pull/2755#discussion_r49932350>:
>
> > @@ -702,7 +765,12 @@ def _predict(X, estimators, is_single_time_sample):
> >      # Compute prediction for each sub-estimator (i.e. per fold)
> >      # if independent, estimators = all folds
> >      for fold, clf in enumerate(estimators):
> > -        _y_pred = clf.predict(X)
> > +        if predict_function == 'predict':
> > +            _y_pred = clf.predict(X)
> > +        elif predict_function == 'predict_proba':
> > +            _y_pred = clf.predict_proba(X)
> > +        elif predict_function == 'decision_function':
> > +            _y_pred = clf.decision_function(X)
>
> @kingjr <https://github.com/kingjr> would you be okay with me making a
> simplified alternative PR?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2755/files#r49932350>.
>